src/drivers/kinetis/uart.c

```diff
@@ -0,0 +1,278 @@
+#include <MK64F12.h>
+#include <drivers/uart.h>
+#include <kernel/kernel_ftab.h>
+#include <queue/queue.h>
+#include <stdbool.h>
+
+
+/**
+ * UNIX-like File IO
+ *     state and prototypes
+ */
+/**
+ * Function implementing write system call for UART
+ * @param uart_context uart context to write to. Configured by uart_init.
+ * @param buf buffer to write
+ * @param bytes number of bytes to write
+ * @return number of bytes actually written. Number of bytes left in buffer
+ * if bytes == 0.
+ */
+unsigned int uart_write(uart_context *context, char *buf, unsigned int bytes);
+
+/**
+ * Function implementing write system call for UART
+ * @param uart_context uart context to read from. Configured by uart_init.
+ * @param buf buffer to read from
+ * @param bytes number of bytes to read
+ * @return number of bytes actually read. Number of bytes available
+ * if bytes == 0.
+ */
+unsigned int uart_read(uart_context *context, char *buf, unsigned int bytes);
+
+/**
+ * Not implemented. UARTs are not de-configured.
+ */
+unsigned int uart_close(uart_context *context);
+/* End UNIX state and prototypes */
+
+/**
+ * Interrupt Service Routines
+ *     state and prototypes
+ */
+/**
+ * Generic ISR for UART RX/TX interrupts
+ */
+void uart_isr(void);
+
+// currently active uart
+static volatile int which_uart;
+// true if any uart is currently being processed by uart_isr
+static volatile bool isr_active;
+
+queue_t txq_headers[6];
+queue_t rxq_headers[6];
+volatile char txqs[6][255];
+volatile char rxqs[6][255];
+uart_context contexts[6] = {

```

Comment:
does every queue get created even if we don't use a specific UART?

Reply:
yes, they have to be statically allocated, as we do not have dynamic memory in the kernel yet.  When we have that, these can go.